syntax = "proto3";

package modyn.model_storage;

service ModelStorage {
  rpc DownloadModel(DownloadModelRequest) returns (DownloadModelRequest) {}
  rpc FetchLatestModel(FetchLatestModelRequest) returns (FetchModelResponse) {}
  rpc FetchModel(FetchModelRequest) returns (FetchModelResponse) {}
}

message DownloadModelRequest {
  int32 training_id = 1;
}

message DownloadModelResponse {
  bool valid = 1;
  string model_path = 2;
}

message FetchLatestModelRequest {
  string model_id = 1;
  optional int32 pipeline_id = 2;
}

message FetchModelRequest {
  string model_id = 1;
  int32 pipeline_id = 2;
  int32 training_id = 3;
}

message FetchModelResponse {
  bool valid = 1;
  string model_path = 2;
}
